Code/Framework/AzToolsFramework/AzToolsFramework/Viewport/LocalViewBookmarkLoader.cpp

```diff
@@ -45,97 +177,174 @@ namespace AzToolsFramework
     }
 
-    static AZStd::string BookmarkSetRegPathAtIndex(const AZStd::string& bookmarkFileName, const int index)
+    static AZStd::string LocalViewBookmarkSetRegPathAtIndex(const AZStd::string& bookmarkFileName, const int index)
     {
-        return AZStd::string::format("%s/%i", BookmarkSetRegPath(bookmarkFileName).c_str(), index);
+        return AZStd::string::format("%s/%i", LocalViewBookmarkSetRegPath(bookmarkFileName).c_str(), index);
     }
 
-    void LocalViewBookmarkLoader::RegisterViewBookmarkLoaderInterface()
+    static AZStd::string LastKnownLocationSetRegPath(const AZStd::string& bookmarkFileName)
     {
-        AZ::Interface<ViewBookmarkLoaderInterface>::Register(this);
+        return "/" + bookmarkFileName + "/LastKnownLocation";
     }
 
-    void LocalViewBookmarkLoader::UnregisterViewBookmarkLoaderInterface()
+    static AZStd::string GenerateBookmarkFileName()
     {
-        AZ::Interface<ViewBookmarkLoaderInterface>::Unregister(this);
-    }
+        auto* prefabEditorEntityOwnershipInterface = AZ::Interface<PrefabEditorEntityOwnershipInterface>::Get();
+        AZ_Assert(prefabEditorEntityOwnershipInterface != nullptr, "PrefabEditorEntityOwnershipInterface is not found.");
+        Prefab::TemplateId rootPrefabTemplateId = prefabEditorEntityOwnershipInterface->GetRootPrefabTemplateId();
 
-    void LocalViewBookmarkLoader::WriteBookmarksSettingsRegistryToFile()
-    {
-        auto registry = AZ::SettingsRegistry::Get();
-        if (!registry)
+        if (rootPrefabTemplateId == Prefab::InvalidTemplateId)
         {
-            AZ_Warning("LocalViewBookmarkLoader", false, "Unable to access global settings registry. Editor Preferences cannot be saved");
-            return;
+            return AZStd::string();
         }
 
-        // resolve path to user project folder
-        const auto localBookmarkFilePath = LocalBookmarkFilePath(m_bookmarkFileName);
+        auto* prefabSystemComponent = AZ::Interface<Prefab::PrefabSystemComponentInterface>::Get();
+        AZ_Assert(
+            prefabSystemComponent != nullptr,
+            "Prefab System Component Interface could not be found. "
+            "It is a requirement for the LocalViewBookmarkLoader class. "
+            "Check that it is being correctly initialized.");
+
+        const auto prefabTemplate = prefabSystemComponent->FindTemplate(rootPrefabTemplateId);
+        const AZStd::string prefabTemplateName = prefabTemplate->get().GetFilePath().Filename().Stem().Native();
+
+        // to generate the file name in which we will store the view Bookmarks we use the name of the prefab + the timestamp
+        // e.g. LevelName_1639763579377.setreg
+        const AZStd::chrono::system_clock::time_point now = AZStd::chrono::system_clock::now();
+        const AZStd::string timeSinceEpoch = AZStd::string::format("%llu", now.time_since_epoch().count());
+        return prefabTemplateName.data() + AZStd::string("_") + timeSinceEpoch + ".setreg";
```

Comment:
nit: Less string operations
````
const AZStd::chrono::system_clock::time_point now = AZStd::chrono::system_clock::now();
return AZStd::string::format("%s_%llu.setreg", prefabTemplateName.data(), now.time_since_epoch().count());
````

Reply:
I didn't change this this time, did you test the above code? 😛 I'll update it next time, it's a nice idea though 👍

Reply:
Saw you did the change. Cheers.